lib/md051.js

```diff
@@ -119,11 +119,22 @@ module.exports = {
         for (const definition of definitions) {
           const { endColumn, startColumn } = definition;
           const text = unescapeStringTokenText(definition);
+          let encodedText = `#${encodeURIComponent(text.slice(1))}`;
+
+          const lineFragmentRe = /^#L\d+$/;
+          if (lineFragmentRe.test(encodedText)) {
+            // eslint-disable-next-line no-continue
+            continue;
+          }
+
+          if (encodedText.startsWith('#L')) {
```

Comment:
I don't understand this part. Why does this need special handling? I'd assume if this were not done then we could move the lineFragmentsRe.test into the if below along with the other checks as I originally proposed. There was no lowercase conversion before, so I think there should not be one after. Maybe this gets to the other point about case sensitivity - I consider the "L111" form to be the only allowed form by this rule since that's what GitHub produces. Anyone using "l111" should switch - which will help avoid confusing letters and numbers. :)

Reply:
Without this condition, the following test case fails:
```md
### L12 ABC

[Valid](#L12-abc)
```

From what I understand, with your previous lowercase related comment, it is to be expected, as the link should have been `[Valid](#l12-abc)` as it is a heading.
But when it's about `#L12`, the L should be uppercase.

Looks fine to me, let's be stricter than GitHub about case sensitivity. But we should maybe document it in `md051.md`? As it might be unexpected for users (me included) that the rule reports errors for valid fragments because of stricter case sensitivity management.

Will simplify the code and make the necessary changes whenever I have time. :+1: